kano_updater/os_version.py

```diff
@@ -13,55 +15,83 @@
 
 class OSVersion(object):
     @staticmethod
+    def _split_helper(os, devstage, number, name=None, variant=None, *args):
+        if args:
+            logger.error("DEV ERROR: Too many values in version string!")
+        return os, devstage, number, name, variant
+
+    @staticmethod
     def from_version_string(vstr):
         try:
-            _os, _codename, _number = vstr.split("-")
+            os, devstage, number, name, variant = OSVersion._split_helper(*vstr.split("-"))
         except:
             msg = "Unknown version string format '{}'".format(vstr)
             raise Exception(msg)
 
-        return OSVersion(_os, _codename, _number)
+        return OSVersion(os, devstage, number, name, variant)
 
     @staticmethod
     def from_version_file(vfile_path):
         with open(vfile_path, 'r') as vfile:
             vstr = vfile.read().strip()
             return OSVersion.from_version_string(vstr)
 
-    def __init__(self, os='Kanux', codename=None, version=None):
+    def __init__(self, os='Kanux', devstage=None, version=None, name=None, variant=None):
         self._os = os
-        self._codename = codename
+        self._devstage = devstage
         self._number = version
+        self._name = name
+        self._variant = variant
 
         try:
             self._major_version = '.'.join(version.split('.')[0:2])
         except:
             self._major_version = version
 
     def to_issue(self):
-        return "{} {} {} \\l".format(self._os, self._codename, self._number)
+        vstr = "{} {} {}".format(self._os, self._devstage, self._number)
+        # Add the name and variant to the string only if they exist.
+        vstr += ' ' + self._name if self._name else ''
```

Comment:
@Ealdwulf To your point from above, yes the function can be called on the old file, hence why here I append and empty string when the `name` cannot be found.

I was careful to preserve backwards compatibility here.